misc-utils/lsfd.c

```diff
@@ -1770,6 +1795,22 @@ int main(int argc, char *argv[])
 		case 'p':
 			parse_pids(optarg, &pids, &n_pids);
 			break;
+		case 'i': {
+			const char *subexpr = NULL;
+			if (optarg) {
+				if (*optarg == '4')
+					subexpr = inet4_subexpr;
```

Comment:
This only looks at the first character.
Allowing things like --inet=4foo.
A strcmp would be better.

Reply:
Updated. Thank you.